tests/FSharp.Compiler.ComponentTests/Conformance/Expressions/ObjectExpressions/ObjectExpressions.fs

```diff
@@ -0,0 +1,318 @@
+            (Error 3860, Line 9, Col 23, Line 9, Col 31, "Static members are not allowed in object expressions.")
+         ]
+         
+    [<FactForNETCOREAPP>]
+    let ``Produce an error when one leaves keyword "static" when implementing static and not static members in an object expression`` () =
+        Fsx """
+type IOperation =
+        static abstract member Execute: unit -> unit
+        abstract member Execute1: unit -> unit
+        static abstract member Execute2: unit -> unit
+        abstract member Execute3: bool
+        
+let _ =
+    { new IOperation with
+        static member Execute() = ()
+        member this.Execute1() = ()
+        static member Execute2() = ()
+        member this.Execute3 = true
+    }
+        """
+         |> withOptions [ "--nowarn:3536" ; "--nowarn:3535" ]
+         |> withLangVersion80
+         |> typecheck
+         |> shouldFail
+         |> withDiagnostics [
+            (Error 3860, Line 10, Col 23, Line 10, Col 30, "Static members are not allowed in object expressions.");
+            (Error 3860, Line 12, Col 23, Line 12, Col 31, "Static members are not allowed in object expressions.");
+         ]
+         
+    [<FactForNETCOREAPP>]
+    let ``No error when implementing only instance members from IWSAM in an object expression`` () =
+        Fsx """
+type ILogger =
+    abstract member Log: string -> unit
+    static abstract member Execute: string -> unit
+        
+let consoleLogger =
+    { new ILogger with
+        member this.Log(message: string) =
+            printfn "%s" message
+    }
+    
+consoleLogger.Log("Hello World")
+        """
+         |> withOptions [ "--nowarn:3536" ; "--nowarn:3535" ]
+         |> withLangVersion80
+         |> compile
+         |> shouldSucceed
+         
+    [<FactForNETCOREAPP>]
+    let ``No error when implementing only instance members from IWSAM(Interface attribute) in an object expression`` () =
+        Fsx """
+[<Interface>]
+type ILogger =
+    abstract member Log: string -> unit
+    static abstract member Execute: string -> unit
+        
+let consoleLogger =
+    { new ILogger with
+        member this.Log(message: string) =
+            printfn "%s" message
+    }
+    
+consoleLogger.Log("Hello World")
+        """
+         |> withOptions [ "--nowarn:3536" ; "--nowarn:3535" ]
+         |> withLangVersion80
+         |> compile
+         |> shouldSucceed
+         
+    [<FactForNETCOREAPP>]
+    let ``No error when implementing only instance members from a type(Interface attribute) in an object expression`` () =
+        Fsx """
+[<Interface>]
+type ILogger =
+    abstract member Log: string -> unit
+    static member Execute x = x
+
+let consoleLogger =
+    { new ILogger with
+        member this.Log(message: string) =
+            printf "%A" message
+    }
+    
+consoleLogger.Log("Hello World")
+        """
+         |> withLangVersion80
+         |> compile
```

Comment:
For last three tests we compile, can we please also run them to see if CLR things the code is fine and can run it?